### ISSUE TITLE
Runs → Print → After n stages for speaker is still generating hundreds of pages

### ISSUE THREAD

**Issue by fvacek** [Quick-Event/quickbox#945]

**Comment by fvacek** [comment]
cannot reproduce after #946

**Comment by paukert** [comment]
Probably same issue as in https://github.com/Quick-Event/quickbox/issues/944. Maybe there is some inconsistency in event data (if it cannot be reproduced) - tested with following qbe files:
- [vcp_letohrad_2023_final - from older QE version](https://github.com/Quick-Event/quickbox/files/12448851/vcp_letohrad_2023_final.zip)
- [vcp_letohrad_2023_final_2 - imported to current QE version](https://github.com/Quick-Event/quickbox/files/12448852/vcp_letohrad_2023_final_2.zip)

```
2023-08-27T21:12:37[reportviewwidget.cpp:82]|I| Show report: C:/QuickEvent/reports/Runs/qml/reports/results_nstagesSpeaker.qml
file:///C:/QuickEvent/reports/Runs/qml/reports/results_nstagesSpeaker.qml:86: TypeError: Cannot read property 'verticalMargin' of undefined
file:///C:/QuickEvent/reports/Runs/qml/reports/results_nstagesSpeaker.qml:85: TypeError: Cannot read property 'horizontalMargin' of undefined
file:///C:/QuickEvent/reports/Runs/qml/reports/results_nstagesSpeaker.qml:84: TypeError: Cannot read property 'pageHeight' of undefined
file:///C:/QuickEvent/reports/Runs/qml/reports/results_nstagesSpeaker.qml:83: TypeError: Cannot read property 'pageWidth' of undefined
```

![obrazek](https://github.com/Quick-Event/quickbox/assets/15065907/6ae279e8-942e-4796-a5c4-76ec431ca064)

**Comment by fvacek** [comment]
are you sure, that you are not using custom reports dir?

**Comment by paukert** [comment]
Tested again, now with AppImage (newest build from Action: https://github.com/Quick-Event/quickbox/actions/runs/5992977484), same problem. Report files have not been modified. Only `After n stages for speaker` and `Competitors with rented cards` are not working. There are some errors in terminal as I already mentioned earlier. I was able to reproduce this issue even with newly created "empty" event.

```
2023-08-29T15:37:19[reportviewwidget.cpp:82]|I| Show report: /tmp/.mount_QuickEf3Ju2q/usr/bin/reports/Runs/qml/reports/startList_classes.qml
2023-08-29T15:37:26[reportviewwidget.cpp:82]|I| Show report: /tmp/.mount_QuickEf3Ju2q/usr/bin/reports/Runs/qml/reports/results_stage.qml
2023-08-29T15:37:30[reportviewwidget.cpp:82]|I| Show report: /tmp/.mount_QuickEf3Ju2q/usr/bin/reports/Runs/qml/reports/competitorsWithCardRent.qml
file:///tmp/.mount_QuickEf3Ju2q/usr/bin/reports/Runs/qml/reports/competitorsWithCardRent.qml:26: TypeError: Cannot read property 'verticalMargin' of undefined
file:///tmp/.mount_QuickEf3Ju2q/usr/bin/reports/Runs/qml/reports/competitorsWithCardRent.qml:25: TypeError: Cannot read property 'horizontalMargin' of undefined
file:///tmp/.mount_QuickEf3Ju2q/usr/bin/reports/Runs/qml/reports/competitorsWithCardRent.qml:24: TypeError: Cannot read property 'pageHeight' of undefined
file:///tmp/.mount_QuickEf3Ju2q/usr/bin/reports/Runs/qml/reports/competitorsWithCardRent.qml:23: TypeError: Cannot read property 'pageWidth' of undefined
2023-08-29T15:39:56[reportviewwidget.cpp:82]|I| Show report: /tmp/.mount_QuickEf3Ju2q/usr/bin/reports/Runs/qml/reports/results_nstagesSpeaker.qml
file:///tmp/.mount_QuickEf3Ju2q/usr/bin/reports/Runs/qml/reports/results_nstagesSpeaker.qml:86: TypeError: Cannot read property 'verticalMargin' of undefined
file:///tmp/.mount_QuickEf3Ju2q/usr/bin/reports/Runs/qml/reports/results_nstagesSpeaker.qml:85: TypeError: Cannot read property 'horizontalMargin' of undefined
file:///tmp/.mount_QuickEf3Ju2q/usr/bin/reports/Runs/qml/reports/results_nstagesSpeaker.qml:84: TypeError: Cannot read property 'pageHeight' of undefined
file:///tmp/.mount_QuickEf3Ju2q/usr/bin/reports/Runs/qml/reports/results_nstagesSpeaker.qml:83: TypeError: Cannot read property 'pageWidth' of undefined
2023-08-29T15:40:05[reportviewwidget.cpp:82]|I| Show report: /tmp/.mount_QuickEf3Ju2q/usr/bin/reports/Runs/qml/reports/results_nstages.qml
2023-08-29T15:40:09[reportviewwidget.cpp:82]|I| Show report: /tmp/.mount_QuickEf3Ju2q/usr/bin/reports/Runs/qml/reports/results_stage.qml
```

**Comment by paukert** [comment]
Could someone else test it please? @arnost00 @otahirs

**Comment by fvacek** [comment]
I can reproduce this error finally, fix is approaching :)

**Comment by arnost00** [comment]
Windows (QE from Actions)[https://github.com/Quick-Event/quickbox/actions/runs/5992977483]
Most of reports looks like normal (with one item or emtpy).
Only `After n stages for speaker` and `Competitors with rented cards` are not working (Show hundreds or thousands of pages).
I've created empty dummy event.

More info in attached log : [quickevent-3.0.5.log](https://github.com/Quick-Event/quickbox/files/12467019/quickevent-3.0.5.log)

**Comment by fvacek** [comment]
fixed in 3.0.6

**Comment by paukert** [comment]
Tested, it's fixed.
